### PR TITLE
feat: rename `Pay` component to `Checkout`

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**
- rename `Pay` component to `Checkout`

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

**Review by fakepixels — APPROVED** [review]

File: src/pay/components/CheckoutButton.tsx [hunk 1]
```diff
@@ -9,16 +9,16 @@ import {
...
+import type { CheckoutButtonReact } from '../types';
+import { usePayContext } from './CheckoutProvider';
 
-export function PayButton({
+export function CheckoutButton({
   className,
   coinbaseBranded,
   disabled,
   icon,
   text = 'Pay',
```
Comment (alessey) [inline src/pay/components/CheckoutButton.tsx:20]: keeping the pay text vs checkout?

File: src/pay/components/CheckoutProvider.tsx [hunk 2]
```diff
@@ -44,13 +44,13 @@ export function usePayContext() {
   return context;
 }
 
-export function PayProvider({
+export function CheckoutProvider({
```
Comment (alessey) [inline src/pay/components/CheckoutProvider.tsx:47]: usePayContext -> useCheckoutContext?

File: src/pay/components/Checkout.test.tsx [hunk 3]
```diff
@@ -41,20 +41,20 @@ describe('Pay', () => {
...
 
   it('should apply the correct className', () => {
     render(
-      <Pay className="test-class">
+      <Checkout className="test-class">
         <div>Test Child</div>
-      </Pay>,
+      </Checkout>,
     );
     const container = screen.getByTestId('pay-provider')
```
Comment (fakepixels) [inline src/pay/components/Checkout.test.tsx:59]: checkout-provider?

File: src/pay/components/Checkout.test.tsx [hunk 4]
```diff
@@ -10,14 +10,14 @@ import {
...
 
 function mock<T>(func: T) {
   return func as Mock;
 }
 
-vi.mock('./PayProvider', () => ({
-  PayProvider: vi.fn(({ children }: { children: React.ReactNode }) => (
+vi.mock('./CheckoutProvider', () => ({
+  CheckoutProvider: vi.fn(({ children }: { children: React.ReactNode }) => (
     <div data-testid="pay-provider">{children}</div>
```
Comment (fakepixels) [inline src/pay/components/Checkout.test.tsx:21]: <div data-testid="checkout-provider">{children}</div>?

**Review by fakepixels — APPROVED** [review]

File: src/pay/components/Checkout.test.tsx (hunk 3 above)
Comment (0xAlec) [reply src/pay/components/Checkout.test.tsx]: done

File: src/pay/components/CheckoutStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import type { CheckoutStatusReact } from '../types';
+
+export function CheckoutStatus({ className }: CheckoutStatusReact) {
+  const { label, labelClassName } = useGetCheckoutStatus();
+
+  return (
+    <div className={cn('flex justify-between', className)}>
+      <div className={text.label2}>
+        <p className={labelClassName}>{label}</p>
+      </div>
```
Comment (cpcramer) [inline src/pay/components/CheckoutStatus.tsx:12]: Can we remove one of these Divs?

File: src/pay/components/CheckoutButton.test.tsx [hunk 6]
```diff
@@ -16,90 +16,90 @@ vi.mock('../../internal/hooks/useIcon', () => ({
...
+    render(<CheckoutButton />);
     const button: HTMLInputElement = screen.getByRole('button');
     expect(button.textContent).toBe('Pay');
     expect(button.disabled).toBe(false);
   });
 
   it('should render Spinner when loading', () => {
-    usePayContextMock.mockReturnValue({
+    useCheckoutContextMock.mockReturnValue({
       lifecycleStatus: { statusName: PAY_LIFECYCLESTATUS.PENDING },
```
Comment (alessey) [inline src/pay/components/CheckoutButton.test.tsx:41]: internal only, but there's a lot of PAY_ consts

File: src/pay/components/CheckoutStatus.tsx (hunk 5 above)
Comment (0xAlec) [reply src/pay/components/CheckoutStatus.tsx:12]: I can try in a follow-up PR, want to keep the scope of this to just renaming the components

File: src/pay/components/CheckoutProvider.test.tsx [hunk 7]
```diff
@@ -522,17 +522,17 @@ describe('PayProvider', () => {
...
-      usePayContext();
+      useCheckoutContext();
       return null;
     };
     const consoleError = vi
       .spyOn(console, 'error')
       .mockImplementation(() => {}); // Suppress error logging
     expect(() => render(<TestComponent />)).toThrow(
-      'usePayContext must be used within a Pay component',
+      'useCheckoutContext must be used within a Pay component',
```
Comment (alessey) [inline src/pay/components/CheckoutProvider.test.tsx:535]: must be used within a Checkout component

File: src/pay/components/CheckoutProvider.tsx [hunk 8]
```diff
@@ -31,26 +31,26 @@ import {
...
+const emptyContext = {} as CheckoutContextType;
+export const CheckoutContext = createContext<CheckoutContextType>(emptyContext);
 
-export function usePayContext() {
-  const context = useContext(PayContext);
+export function useCheckoutContext() {
+  const context = useContext(CheckoutContext);
   if (context === emptyContext) {
-    throw new Error('usePayContext must be used within a Pay component');
+    throw new Error('useCheckoutContext must be used within a Pay component');
```
Comment (alessey) [inline src/pay/components/CheckoutProvider.tsx:42]: must be used within a Checkout component

File: src/pay/constants.ts [hunk 9]
```diff
@@ -16,24 +16,24 @@ export const PAY_INVALID_PARAMETER_ERROR_MESSAGE =
 export const UNCAUGHT_PAY_ERROR_MESSAGE = 'UNCAUGHT_PAY_ERROR';
```
Comment (alessey) [inline src/pay/constants.ts:16]: uncaught checkout error?

**Pull request merged**